parser/parser.go

```diff
@@ -1192,6 +1199,50 @@ func (p *Parser) parseGo() stmt.Stmt {
 	return g
 }
 
+func (p *Parser) parseConst() *stmt.Const {
+	s := &stmt.Const{
+		Position: p.pos(),
+	}
+items:
+	for {
+		p.expect(token.Ident)
+		s.NameList = append(s.NameList, p.s.Literal.(string))
+		p.next()
+		switch p.s.Token {
+		case token.Ident:
+			s.Type = p.parseType()
+			if p.s.Token == token.Assign {
+				p.next()
+				s.Values = p.parseExprs()
+			}
+			if p.s.Token == token.Semicolon {
+				break items
+			}
+		case token.Comma:
+			p.next()
+			continue
+		case token.Assign:
+			p.next()
+			s.Values = p.parseExprs()
+			break items
+		case token.Semicolon:
+			break items
+		default:
+			p.errorf("invalid token %v in const declaration", p.s.Token)
```

Comment:
Need a break here, otherwise parsing the file "const ( x" will get stuck.

Reply:
done.